src/group_theory/order_of_element.lean

```diff
@@ -46,109 +113,72 @@ variables {H : Type u} [add_monoid H] {x : H}
 
 open_locale classical
 
-/-- `add_order_of h` is the additive order of the element `x`, i.e. the `n ≥ 1`, s. t. `n •ℕ x = 0`
-if it exists. Otherwise, i.e. if `x` is of infinite order, then `add_order_of x` is `0` by
-convention.-/
+/-- `add_order_of x` is the order of the element `x`, i.e. the `n ≥ 1`, s.t. `n •ℕ x = 0` if it
+exists. Otherwise, i.e. if `x` is of infinite order, then `add_order_of x` is `0` by convention.-/
 noncomputable def add_order_of (x : H) : ℕ :=
-if h : ∃ n, 0 < n ∧ n •ℕ x = 0 then nat.find h else 0
+minimal_period (+ x) 0
```

Comment:
```suggestion
minimal_period ((+) x) 0
```
might be nicer to work with, as that binds the first argument of add not the second, and so results no lambda binders.

Reply:
That change broke everything and it was not immediately clear to me how to fix it.

Reply:
Right, you'd need to change all the proofs too - it comes down to where you want to define things in terms of iterated left or right multiplication

Reply:
I'm not sure that I know whether this is Eric's suggestion, but I would prefer to have the "free" variable to be the one on the right: my recollection is that (\la x, a * x) behaves better than (\la x, x * a).  However, if Eric's thinks differently, then I would go with Eric's preference!

Reply:
You clearly have more experience, why would the version with the brackets be better?

Reply:
There is a specific lemma is the file `algebra/regular` where for the `right` version the proof is the "same" as the `left` version, except that you need an extra type annotation, since the output is garbled, although defeq to what you want.  This is enough to mess up a `rw`, though.

Reply:
`((+) x)` desugars to `has_add.add x`, while `(+ x)` desugars to `(λ a, has_add.add a x)`. The latter can sometimes be more awkward because `(+ x) y` is `(λ a, has_add.add a x) y`, while `((+) x y)` is the simpler `has_add.add x y`.

Which you pick doesn't really matter that much, but I wanted to make sure you were aware that you were making the choice. The inconvenience of the `λ`s outweighs the parentheses needed, but probably doesn't outweigh any mathematical justification you might have.

Reply:
The only mathematical justification I could think of would be if one wants to apply the concept of order to non-associative structures, but I would have to think which one would be preferrable in this case. I didn't have a mathematical justifcation, maybe Yury who suggested it in #6770, has one.

Reply:
Eric, I think that I am with you: I would prefer to use
* `minimal_period ((+) x) 0`, and `minimal_period ((*) x) 0`
* as opposed to the current `minimal_period (+ x) 0` and  `minimal_period (* x) 0`.

However, as Julian pointed out, this means refactoring a bunch of previous definitions that use the opposite convention.  I do not know whether there was a reason to prefer the current convention, as opposed to the other one.  In any case, It may not be a huge refactor to make the change, but I would like to confirm that this does not go against an explicit choice that was already made!

Reply:
For me, it doesn't matter if we can get the other version running, that's fine with me. But I also didn't encounter problems with the lambda-version.